src/main/java/com/jcraft/jsch/KeyExchange.java

```diff
@@ -290,187 +325,172 @@ protected byte[] normalize(byte[] secret) {
-      tmp = new byte[j];
-      System.arraycopy(K_S, i, tmp, 0, j);
-      i += j;
-      p = tmp;
-      j = ((K_S[i++] << 24) & 0xff000000) | ((K_S[i++] << 16) & 0x00ff0000)
-          | ((K_S[i++] << 8) & 0x0000ff00) | ((K_S[i++]) & 0x000000ff);
-      tmp = new byte[j];
-      System.arraycopy(K_S, i, tmp, 0, j);
-      i += j;
-      q = tmp;
-      j = ((K_S[i++] << 24) & 0xff000000) | ((K_S[i++] << 16) & 0x00ff0000)
-          | ((K_S[i++] << 8) & 0x0000ff00) | ((K_S[i++]) & 0x000000ff);
-      tmp = new byte[j];
-      System.arraycopy(K_S, i, tmp, 0, j);
-      i += j;
-      g = tmp;
-      j = ((K_S[i++] << 24) & 0xff000000) | ((K_S[i++] << 16) & 0x00ff0000)
-          | ((K_S[i++] << 8) & 0x0000ff00) | ((K_S[i++]) & 0x000000ff);
-      tmp = new byte[j];
-      System.arraycopy(K_S, i, tmp, 0, j);
-      i += j;
-      f = tmp;
-
-      SignatureDSA sig = null;
-      try {
-        Class<? extends SignatureDSA> c =
-            Class.forName(session.getConfig("signature.dss")).asSubclass(SignatureDSA.class);
-        sig = c.getDeclaredConstructor().newInstance();
-        sig.init();
-      } catch (Exception e) {
-        throw new JSchException(e.toString(), e);
-      }
-      sig.setPubKey(f, p, q, g);
-      sig.update(H);
-      result = sig.verify(sig_of_H);
+      byte[] serverPublicKeyByteArray = certificate.getCertificatePublicKey();
+      buffer = new OpenSshCertificateBuffer(serverPublicKeyByteArray);
+      alg = buffer.getString(UTF_8);
+      this.hostKeyCertificate = certificate;
+    }
 
-      if (session.getLogger().isEnabled(Logger.INFO)) {
-        session.getLogger().log(Logger.INFO, "ssh_dss_verify: signature " + result);
-      }
-    } else if (alg.equals("ecdsa-sha2-nistp256") || alg.equals("ecdsa-sha2-nistp384")
-        || alg.equals("ecdsa-sha2-nistp521")) {
-      byte[] tmp;
-      byte[] r;
-      byte[] s;
-
-      // RFC 5656,
-      type = ECDSA;
-      key_alg_name = alg;
-
-      j = ((K_S[i++] << 24) & 0xff000000) | ((K_S[i++] << 16) & 0x00ff0000)
-          | ((K_S[i++] << 8) & 0x0000ff00) | ((K_S[i++]) & 0x000000ff);
-      tmp = new byte[j];
-      System.arraycopy(K_S, i, tmp, 0, j);
-      i += j;
-      j = ((K_S[i++] << 24) & 0xff000000) | ((K_S[i++] << 16) & 0x00ff0000)
-          | ((K_S[i++] << 8) & 0x0000ff00) | ((K_S[i++]) & 0x000000ff);
-      i++;
-      tmp = new byte[(j - 1) / 2];
-      System.arraycopy(K_S, i, tmp, 0, tmp.length);
-      i += (j - 1) / 2;
-      r = tmp;
-      tmp = new byte[(j - 1) / 2];
-      System.arraycopy(K_S, i, tmp, 0, tmp.length);
-      i += (j - 1) / 2;
-      s = tmp;
-
-      SignatureECDSA sig = null;
-      try {
-        Class<? extends SignatureECDSA> c =
-            Class.forName(session.getConfig(alg)).asSubclass(SignatureECDSA.class);
-        sig = c.getDeclaredConstructor().newInstance();
-        sig.init();
-      } catch (Exception e) {
-        throw new JSchException(e.toString(), e);
-      }
+    switch (alg) {
```

Comment:
Can you revert this back to using the if, else if chain instead of converting it to a switch statement?
This makes it much more difficult for me to review and confirm that no logic has been accidentally altered.